### ISSUE TITLE
Support typeof tuple[number] pattern

### ISSUE THREAD

**Issue by cometkim** [vedantroy/typecheck.macro#20]
```ts
// Use this to get an iterable of possible values.
const namingConvention = ['snake_case', 'camelCase'] as const;

type NamingConvention = typeof namingConvention[number];

registerType('NamingConvention');

export const isValidNamingConvention = createValidator<NamingConvention>();
```

got error:

```
Unexpected error because TSIndexedAccessType was not expected. Please report this to the developer.
```
